src/log.py

```diff
@@ -0,0 +1,55 @@
+"""
+Data Log
+
+By: Norma Chang
+"""
+import json
+import time
+
+def log(data, data_type):
+    """
+    Logs dictionary data from leader
+    """
+    if data_type == 'leader':
+        leader_dict = {
+            'Time' : time.time(), 
+            'Input' : data._distance,
+            'Speed' : data._speed,
+        }
+        write_to_file(leader_dict, data_type)
+    elif data_type == 'follower':
+        follower_dict = {
+            'Time' : time.time(), 
+            'Distance' : data._distance,
+            'Speed' : data._speed,
+            'Turn Angle' : data._turn_angle,
+            'Yaw' : data._yaw,
+        }
+        write_to_file(follower_dict, data_type)
```

Comment:
Why are you passing the vehicle role as the file name?

Reply:
so we'll know which log is the leader and which is the follower.

Reply:
It can be done by storing the file name automatically when it's generated at object instantiation. It will then use the same name to write any dictionary we pass into it.